### PR TITLE
fixes #4

### LINKED ISSUE #4: Wrong query building when fetching with `withRelated`
I came across this issue running queries with option `withRelated` and polymorphic props. The queries executed against the db end up having their "deleted_at" checks mixed up and break.

In my scenario there are 3 entities: `girls`, `boys`, `parents`. `parents` have a `child` prop that can either be a girl or a boy.

```js
const Boy = bookshelf.Model.extend({
  tableName: 'boys',
  softDelete: true
})
const Girl = bookshelf.Model.extend({
  tableName: 'girls',
  softDelete: true
})
const Parent = bookshelf.Model.extend({
  tableName: 'parents',
  child () {
    return this.morphTo('child', Boy, Girl)
  },
  softDelete: true
})
```

Then, have to create a boy, a parent having that boy, a girl and a parent having that girl:

```js
const waterfall = require('promise-waterfall')

function createBoy() {
  return Boy.forge().save().then(function (child) {
    return child.id
  })
}
function createGirl() {
  return Girl.forge().save().then(function (child) {
    return child.id
  })
}
function createParent() {
  return Parent.forge().save().then(function (parent) {
    return parent.id
  })
}
function setChild(parentId, childId, type) {
  return Parent.forge({
    id: parentId
  }).save({
    child_id: childId,
    child_type: type
  })
}

const tasks = waterfall([
  function () {
    return createBoy()
  },
  function (childId) {
    return createParent().then(function (parentId) {
      return setChild(parentId, childId, 'boys')
    })
  },
  function () {
    return createGirl()
  },
  function (parentBoyId) {
    return createParent().then(function (parentId) {
      return setChild(parentId, parentBoyId, 'girls')
    })
  }
])
```

Finally, fetch all parents with their children:

```js
tasks.then(function () {
  return Parent.fetchAll({
    withRelated: ['child'],
    debug: true
  })
})
```

The output is:

```
{ method: 'select',
  options: {},
  timeout: false,
  bindings: [],
  sql: 'select "parents".* from "parents" where "parents"."deleted_at" is null and "parents"."deleted_at" is null' }
{ method: 'select',
  options: {},
  timeout: false,
  bindings: [ '1' ],
  sql: 'select "boys".* from "boys" where "id" in (?)' }
{ method: 'select',
  options: {},
  timeout: false,
  bindings: [ '1' ],
  sql: 'select "girls".* from "girls" where "id" in (?) and "boys"."deleted_at" is null and "girls"."deleted_at" is null' }
Err: error: select "girls".* from "girls" where "id" in ($1) and "boys"."deleted_at" is null and "girls"."deleted_at" is null - missing FROM-clause entry for table «boys»
```

As can be seen in the logs, the `and "boys"."deleted_at" is null` in the last query (girls) is wrong and breaking. In addition, that condition is missing from the second query (boys).

Tested using:
- bookshelf@0.9.5
- bookshelf-paranoia@0.7.0
- knex@0.11.5

Will really appreciate any help with the issue. Thanks!!!

PS: You can see the whole test file at [test-bookshelf-paranoia.js](https://gist.github.com/gabmontes/3f7723a36fc3bec622252fd470da15fe)

### PR DESCRIPTION
**Pull request by alanhoff**

### CODE REVIEW

**Comment by alanhoff** [comment]
@bsiddiqui ready for review

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "bookshelf": "^0.9.4",
     "co": "^4.6.0",
     "code": "^2.2.0",
+    "eslint": "^2.13.1",
```
Comment (bsiddiqui) [inline package.json]: @alanhoff did you mean to add this?

Comment (alanhoff) [inline package.json]: I was using node 6 and it's a peer dep from standard that isn't installed automatically, will do no harm here

**Comment by bsiddiqui** [comment]
LGTM

**Pull request merged**